process/transaction/interceptedTransaction.go

```diff
@@ -230,8 +242,9 @@ func (inTx *InterceptedTransaction) processFields(txBuff []byte) error {
 
 // integrity checks for not nil fields and negative value
 func (inTx *InterceptedTransaction) integrity(tx *transaction.Transaction) error {
-	if tx.Version < inTx.minTransactionVersion {
-		return process.ErrInvalidTransactionVersion
+	err := inTx.txOptionChecker.CheckTxVersion()
```

Comment:
missing unit tests for the new functionality. The unit tests will be done in a more simpler manner if the tx version checker will be injected and a new interface + mock will be added.